### PR TITLE
Allow users to specify http/https for accessing nodes

### PR DESCRIPTION
**Pull request by mightybyte**

### CODE REVIEW

File: src/Commands/Local.hs [hunk 1]
```diff
@@ -36,15 +36,17 @@ localCommand e args = do
       bss <- mapM LB.readFile fs
       res <- runExceptT $ do
         allTxs <- hoistEither $ first unlines $ parseAsJsonOrYaml bss
-        hpPairs <- handleOptionalNode e allTxs $ _nodeTxCmdArgs_node args
-        forM hpPairs $ \(hp, txs) -> do
-          n <- ExceptT $ getNode le hp
+        shpPairs <- handleOptionalNode e allTxs $ _nodeTxCmdArgs_node args
+        forM shpPairs $ \(shp, txs) -> do
+          logEnv e DebugS $ logStr "Here A"
```
Comment (jwiegley) [inline src/Commands/Local.hs:41]: I assume this were for debugging and should be removed now?

File: src/Types/HostPort.hs [hunk 2]
```diff
@@ -24,12 +25,50 @@ hostPortFromText t = do
...
+
+data Scheme = Http | Https
+  deriving (Eq,Ord,Show,Read,Generic)
+
+schemeText :: Scheme -> Text
+schemeText Https = "https"
+schemeText Http = "http"
+
+data SchemeHostPort = SchemeHostPort
+  { _shp_scheme :: Maybe Scheme
```
Comment (jwiegley) [inline src/Types/HostPort.hs:46]: What does a scheme of None mean?

File: src/Types/Node.hs [hunk 3]
```diff
@@ -49,39 +46,144 @@ data Node = Node
...
-                if statusIsSuccessful (responseStatus resp3)
-                  then return $ Right $ Node Https h httpMgr PactServer Nothing
-                  else return $ Left ("Error requesting from " <> infoUrl)
+getNodeServiceApi :: LogEnv -> SchemeHostPort -> IO (Either String Node)
+getNodeServiceApi le shp = do
+    -- TODO Add exception handling from below comment & delete comment
+
+    let hp = _shp_hostPort shp
+    case _shp_scheme shp of
+      Nothing -> runExceptT $
```
Comment (jwiegley) [inline src/Types/Node.hs:55]: Ah, it means auto-select. Maybe better to have that as an explicit choice of Scheme?

File: src/Commands/Local.hs (hunk 1 above)
Comment (mightybyte) [reply src/Commands/Local.hs:41]: Ahh yes.  Will update.

File: src/Types/HostPort.hs (hunk 2 above)
Comment (mightybyte) [reply src/Types/HostPort.hs:46]: It means that the user didn't specify, which gives a better user experience by allowing any of the following to work from the user:

```
localhost
localhost:4443
https://localhost:4443
http://localhost
http://localhost:1848
```

If the user wants to be explicit they can be and if not, then kda-tool can attempt to figure it out in whatever manner is appropriate for the context (which might be different for different commands).

File: src/Types/Node.hs (hunk 3 above)
Comment (mightybyte) [reply src/Types/Node.hs:55]: The main thing this PR does is introduce the ability for the user to choose explicitly while also keeping support for not specifying to preserve backwards compatibility and smooth the user experience.

File: src/Types/HostPort.hs (hunk 2 above)
Comment (jwiegley) [reply src/Types/HostPort.hs:46]: Makes sense

**Pull request merged**